pets/zookeeper/zookeeper.yaml

```diff
@@ -0,0 +1,112 @@
+# A headless service to create DNS records
+apiVersion: v1
+kind: Service
+metadata:
+  annotations:
+    pod.alpha.kubernetes.io/tolerate-unready-endpoints: "true"
+  name: zk
+  labels:
+    app: zk
+spec:
+  ports:
+  - port: 2888
+    name: peer
+  - port: 3888
+    name: leader-election
+  # *.zk.default.svc.cluster.local
+  clusterIP: None
+  selector:
+    app: zk
+---
+apiVersion: apps/v1alpha1
+kind: PetSet
+metadata:
+  name: zoo
+spec:
+  serviceName: "zk"
+  replicas: 3
+  template:
+    metadata:
+      labels:
+        app: zk
+      annotations:
+        pod.alpha.kubernetes.io/initialized: "true"
+        pod.alpha.kubernetes.io/init-containers: '[
+            {
+                "name": "install",
+                "image": "bprashanth/zookeeper-install:0.1",
+                "imagePullPolicy": "Always",
+                "args": ["--version=3.5.0-alpha", "--install-into=/opt", "--config-scripts=/config-scripts"],
```

Comment:
Clever.  You should use workdir as well for one of the directories (maybe config scripts).